src/main/webapp/js/feed-loader.js

```diff
@@ -0,0 +1,51 @@
+
+// Fetch messages and add them to the page.
+function fetchMessages(){
+  const url = '/feed';
+  fetch(url).then((response) => {
+    return response.json();
+  }).then((messages) => {
+    const messageContainer = document.getElementById('message-container');
+    if(messages.length == 0){
+      messageContainer.innerHTML = '<p>There are no posts yet.</p>';
+    }
+    else{
+      messageContainer.innerHTML = '';
+    }
+    messages.forEach((message) => {
+      const messageDiv = buildMessageDiv(message);
+      messageContainer.appendChild(messageDiv);
+    });
+  });
+}
+
+function buildMessageDiv(message){
```

Comment:
I think this function also shows up in the user page loader, right? If you're feeling especially brave you might consider moving this into a shared location?